packages/block-library/src/navigation/edit/menu-inspector-controls.js

```diff
@@ -150,6 +152,27 @@ const MenuInspectorControls = ( props ) => {
 					>
 						{ __( 'Menu' ) }
 					</Heading>
+					<BlockControls group="inline">
+						<NavigationMenuSelector
+							currentMenuId={ currentMenuId }
+							onSelectClassicMenu={ onSelectClassicMenu }
+							onSelectNavigationMenu={ onSelectNavigationMenu }
+							onCreateNew={ onCreateNew }
+							createNavigationMenuIsSuccess={
+								createNavigationMenuIsSuccess
+							}
+							createNavigationMenuIsError={
+								createNavigationMenuIsError
+							}
+							actionLabel={ actionLabel }
+							isManageMenusButtonDisabled={
+								isManageMenusButtonDisabled
+							}
+							icon={ chevronDown }
+							text={ __( 'Change' ) }
```

Comment:
I was envisaging show the name of the menu here truncated so as not to be overly long. Something like this...

<img width="621" alt="Screen Shot 2023-09-08 at 11 59 19" src="https://github.com/WordPress/gutenberg/assets/444434/699b689e-8966-4116-9cd2-5c8abc4236ca">